cudapoa/src/cudapoa_batch.cuh

```diff
@@ -502,11 +502,23 @@ protected:
         }
         else
         {
-            // Verify that weightsw are positive.
+            // Verify that weights are positive.
+            bool all_base_weights_are_zero = true;
             for (int32_t i = 0; i < seq_len; i++)
             {
                 throw_on_negative(weights[i], "Base weights need to be non-negative");
+                if (weights[i] > 0)
+                {
+                    all_base_weights_are_zero = false;
+                }
+            }
+
+            // all base weights can not be 0
+            if (all_base_weights_are_zero)
+            {
+                throw std::invalid_argument("All base weights can not be zero");
```

Comment:
let's return an error code here like "cudapoa_zero_weighted_sequence" instead of throwing an exception. that way these sequences can be skipped by the user if they so choose

Reply:
yes, that way is less interruptive. My initial thought was if returned as error code it may remain hidden, since we add it to the status vector in `Batch::add_poa_group(vector<StatusType>& per_seq_status,  const Group& poa_group)`, but return status of  `StatusType Batch::add_poa_group()` will be `success`, even if all sequences to be added to the group fail.
as a workaround, I added a couple of error codes, `zero_weighted_poa_sequence` and `empty_poa_group`

Reply:
awesome, I like this!